lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -76,6 +77,21 @@ public function register_routes() {
 	}
 
 	/**
+	 * Permissions check for getting all users.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		// Check if roles is specified in GET request and if user can list users.
+		if ( ! empty( $request['roles'] ) && ! current_user_can( 'list_users' ) ) {
+			return new WP_Error( 'rest_user_cannot_view', __( 'Sorry, you cannot filter by role.' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
I don't know haha, I copied it from some of the other permission checks on get requests. For example, if someone put in context=edit they would get this error code if unauthenticated/unauthorized.